spring-web/src/main/java/org/springframework/http/ResponseEntity.java

```diff
@@ -567,6 +577,12 @@ public BodyBuilder contentType(MediaType contentType) {
 			return this;
 		}
 
+		@Override
+		public BodyBuilder contentDisposition(ContentDisposition contentDisposition) {
+			this.headers.setContentDisposition(contentDisposition);
```

Comment:
This looks well written, Just a small comment 
Consider adding a null check for the contentDisposition parameter. If null values are not allowed, you could throw an IllegalArgumentException